base/src/main/java/proguard/InputReader.java

```diff
@@ -231,6 +236,27 @@ public void readInput(String          messagePrefix,
         }
     }
 
+    // TODO: For debugging
+    public static void main(String[] args) throws IOException {
+        Set<String> withJmods = new HashSet<>();
+        File file = new File("jdk-25-with-jmods/jmods");
+        new InputReader(new Configuration()).readInput("message", new ClassPathEntry(file, false), dataEntry -> withJmods.add(dataEntry.getName()));
+
+        file = new File("jdk-25-without-jmods/jmods");
+        Set<String> withoutJmods = new HashSet<>();
+        new InputReader(new Configuration()).readInput("message", new ClassPathEntry(file, false), dataEntry -> withoutJmods.add(dataEntry.getName()));
+
+        Set<String> missing = new HashSet<>(withJmods);
+        missing.removeAll(withoutJmods);
+        Set<String> unexpected = new HashSet<>(withoutJmods);
+        unexpected.removeAll(withJmods);
+
+        System.out.println("=== MISSING ===");
+        missing.forEach(System.out::println);
+
+        System.out.println("=== UNEXPECTED ===");
+        unexpected.forEach(System.out::println);
```

Comment:
I noticed some differences here for non-`.class` files. Not sure if that is a problem. For exampe the jmods seem to have many of them nested within `classes/`, whereas for `jrt:/` they seem to be at the top level of a module.